src/AppActivationCodes.test.tsx

```diff
@@ -0,0 +1,87 @@
+import React from 'react'
+import { fireEvent, render, wait } from 'react-testing-library'
+
+import electionSample from './data/electionSample.json'
+
+import App, { electionKey, mergeWithDefaults } from './App'
+import { Election } from './config/types'
+
+const electionSampleAsString = JSON.stringify(
+  mergeWithDefaults(electionSample as Election)
+)
+
+beforeEach(() => {
+  window.localStorage.clear()
+  window.location.href = '/'
+})
+
+it(`Activation code is reset if not submitted`, async () => {
+  window.localStorage.setItem(electionKey, electionSampleAsString)
+  const { getByTestId } = render(<App />)
+
+  jest.useFakeTimers()
+
+  const activationCodeInput = getByTestId('activation-code') as HTMLInputElement
+
+  // Set invalid activation code and do not submit form
+  fireEvent.change(getByTestId('activation-code'), {
+    target: { value: 'VX.precinct-21.5R' },
+  })
+  expect(activationCodeInput.value).toBe('VX.precinct-21.5R')
+
+  // Activation code input value is cleared.
+  jest.runAllTimers()
```

Comment:
in case we later have `setInterval` used in other functions (ahem, I think I know of one), I recommend `runOnlyPendingTimers` as the default approach to testing.